src/main/java/com/epam/aidial/core/controller/ControllerSelector.java

```diff
@@ -158,38 +158,46 @@ private static Controller selectPost(Proxy proxy, ProxyContext context, String p
         match = match(PATTERN_RATE_RESPONSE, path);
         if (match != null) {
             String deploymentId = match.group(1);
-            RateResponseController controller = new RateResponseController(proxy, context);
-            return () -> controller.handle(deploymentId);
+
+            Function<Deployment, String> getter = (model) -> {
+                return Optional.ofNullable(model)
+                    .map(d -> d.getFeatures())
+                    .map(t -> t.getRateEndpoint())
+                    .orElse(null);
+            };
+
+            DeploymentFeatureController controller = new DeploymentFeatureController(proxy, context);
```

Comment:
The controller behaves differently than the original one:

- Don't log request body in case if there is no rate endpoint  provided
- Return 403 if rate endpoint is not provided in the config but should return 200. 403 should be returned if deployment is not found or not eligible.